cpu/samd5x/cpu.c

```diff
@@ -41,21 +40,29 @@ static void dfll_init(void)
 
     OSCCTRL->DFLLCTRLB.reg = reg;
     OSCCTRL->DFLLCTRLA.reg = OSCCTRL_DFLLCTRLA_ENABLE;
+
+    while (!OSCCTRL->STATUS.bit.DFLLRDY) {}
 }
-#endif
 
 static void fdpll0_init(uint32_t f_cpu)
 {
+    /* We source the DPLL from 32kHz GCLK1 */
     const uint32_t LDR = ((f_cpu << 5) / 32768);
 
+    /* disable the DPLL before changing the configuration */
+    OSCCTRL->Dpll[0].DPLLCTRLA.bit.ENABLE = 0;
+    while (OSCCTRL->Dpll[0].DPLLSYNCBUSY.reg) {}
+
+    /* set DPLL clock source */
     GCLK->PCHCTRL[OSCCTRL_GCLK_ID_FDPLL0].reg = GCLK_PCHCTRL_GEN(1) | GCLK_PCHCTRL_CHEN;
+    while (!(GCLK->PCHCTRL[OSCCTRL_GCLK_ID_FDPLL0].reg & GCLK_PCHCTRL_CHEN)) {}
 
     OSCCTRL->Dpll[0].DPLLRATIO.reg = OSCCTRL_DPLLRATIO_LDRFRAC(LDR & 0x1F)
-                       | OSCCTRL_DPLLRATIO_LDR((LDR >> 5) - 1);
+                                   | OSCCTRL_DPLLRATIO_LDR((LDR >> 5) - 1);
 
+    /* Without LBYPASS, startup takes very long, reson unknown. */
     OSCCTRL->Dpll[0].DPLLCTRLB.reg = OSCCTRL_DPLLCTRLB_REFCLK_GCLK
-                       | OSCCTRL_DPLLCTRLB_WUF
```

Comment:
I would be better to not remove this regarding errata stuff.

Reply:
It's back - with the errata now it has a proper justification to be there.